matter_server/client/connection.py

```diff
@@ -80,13 +80,16 @@ async def connect(self) -> None:
         self.server_info = info
 
         # basic check for server schema version compatibility
-        if info.min_supported_schema_version > SCHEMA_VERSION:
+        if info.min_supported_schema_version > SCHEMA_VERSION > info.schema_version:
```

Comment:
How can the minimum supported schema version be higher than the latest schema version of the server? That should never happen. We need two conditions and an `or` operator.

```
Python 3.12.4 (main, Jun  8 2024, 18:29:57) [GCC 9.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.22.2 -- An enhanced Interactive Python. Type '?' for help.

In [1]: 1 > 2 > 3
Out[1]: False

In [2]: 3 > 2 > 1
Out[2]: True

In [3]: 1 > 2 > 1
Out[3]: False

In [4]: 2 > 1 > 2
Out[4]: False

In [5]: 1 > 3 > 2
Out[5]: False

In [6]: 1 > 3 or 3 > 2
Out[6]: True
```

Reply:
Hm, whoops yeah butchered that one :see_no_evil: 

I wonder if it would be better to just have two exceptions `ServerVersionTooNew`, `ServerVersionTooOld` instead of `InvalidServerVersion`? :thinking:

Reply:
I went down this road: There are two Exception now deriving from `InvalidServerVersion`.